### ISSUE TITLE
False positive IDE0051 and IDE0052, regression from 10.0.103 and 10.0.104.

### ISSUE THREAD

**Issue by PetSerAl** [dotnet/roslyn#82704]
**Version Used**:
10.0.200
10.0.104
10.0.103

**Steps to Reproduce**:
Class.cs:
```csharp
namespace Test;

public static class Class
{
	extension(string) {
		public static string T() => Test1();
		public static string F() => Field1;
	}
	private static string Test1() => "Test1";
	private static string Test2() => "Test2";
	private static string Field1 = string.Empty;
	private static string Field2 = string.Empty;
}
```
Test.csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<EnforceCodeStyleInBuild>True</EnforceCodeStyleInBuild>
		<TargetFramework>net10.0</TargetFramework>
		<ImplicitUsings>enable</ImplicitUsings>
		<Nullable>enable</Nullable>
	</PropertyGroup>
</Project>
```
.editorconfig
```
root = true
[*.cs]
dotnet_diagnostic.IDE0051.severity = warning
dotnet_diagnostic.IDE0052.severity = warning
```

**Diagnostic Id**:
10.0.103 and 10.0.104 report 2 diagnostics:
warning IDE0051: Private member 'Class.Test2' is unused
warning IDE0052: Private member 'Class.Field2' can be removed as the value assigned to it is never read

10.0.200 report 4 diagnostics:
warning IDE0051: Private member 'Class.Test1' is unused
warning IDE0051: Private member 'Class.Test2' is unused
warning IDE0052: Private member 'Class.Field1' can be removed as the value assigned to it is never read
warning IDE0052: Private member 'Class.Field2' can be removed as the value assigned to it is never read

**Expected Behavior**:
10.0.200 should not report false positive diagnostics.

**Actual Behavior**:
10.0.200 report false positive diagnostics.
